### PR TITLE
KHWE-5 implement albums view

### PR DESCRIPTION
**Pull request by OlegShynkarenko**
screencapture-localhost-8080-2018-11-01-01_35_17

### CODE REVIEW

File: src/components/AlbumsComponents/AlbumListItemComponent/AlbumListItemComponent.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+import AlbumListItem from "./AlbumListItem.html";
+
+export class AlbumListItemComponent {
+  constructor(mountPoint) {
+    this.mountPoint = mountPoint;
+  }
+
+  initMaterialCard() {
+    const cardRipple = document.querySelectorAll(".mdc-card__primary-action");
+    console.log(cardRipple);
```
Comment (ghost) [inline src/components/AlbumsComponents/AlbumListItemComponent/AlbumListItemComponent.js]: Remove console.log() pls

File: src/components/AlbumsComponents/AlbumListItemComponent/AlbumListItemComponent.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    Array.from(cardRipple).forEach(el => {
+      new MDCRipple(el);
+      console.log(el);
```
Comment (ghost) [inline src/components/AlbumsComponents/AlbumListItemComponent/AlbumListItemComponent.js]: same

File: src/components/AlbumsComponents/AlbumListItemComponent/AlbumListItemComponent.js (hunk 1 above)
Comment (OlegShynkarenko) [reply src/components/AlbumsComponents/AlbumListItemComponent/AlbumListItemComponent.js]: Removed

File: src/components/AlbumsComponents/AlbumListItemComponent/AlbumListItemComponent.js (hunk 2 above)
Comment (OlegShynkarenko) [reply src/components/AlbumsComponents/AlbumListItemComponent/AlbumListItemComponent.js]: Removed

**Review by pavlobaranchuk — APPROVED** [review]
Hi Oleg. Looks great. Could you please move all images to our "assets" folder on a "./src" level?

**Comment by ihor4568** [comment]
You have problems with scrolling 

image


sidebar should not scroll

**Comment by ihor4568** [comment]
Please fix grid

image

**Comment by ihor4568** [comment]
Please use pictures from real albums - and real  album names and artists - not just "album name"

File: src/components/Albums/albumsComponent.scss [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  padding-top: 0.6rem;
+  max-width: 100rem;
+  margin: 0 auto;
+
+  &__heading {
+    margin-left: 2rem;
+  }
+
+  &__list {
+    //padding-left: 0.6rem;
```
Comment (ihor4568) [inline src/components/Albums/albumsComponent.scss]: why comment here?

File: src/components/Albums/AlbumListItemComponent/AlbumListItem.html [hunk 4]
```diff
@@ -0,0 +1,156 @@
+<div class="albums__card mdc-card mdc-card-album">
```
Comment (ihor4568) [inline src/components/Albums/AlbumListItemComponent/AlbumListItem.html]: please use loop or foreach here - you can use ejs

File: src/components/Albums/AlbumListComponent/AlbumList.html [hunk 5]
```diff
@@ -0,0 +1,6 @@
+<section class="main__albums">
```
Comment (ihor4568) [inline src/components/Albums/AlbumListComponent/AlbumList.html]: why is it here? where is BEM?

File: src/components/Albums/AlbumListItemComponent/AlbumListItemComponent.js [hunk 6]
```diff
@@ -0,0 +1,26 @@
+import { MDCRipple } from "@material/ripple";
+import albumListItemTemplate from "./AlbumListItem.html";
+
+export class AlbumListItemComponent {
+  constructor(mountPoint) {
+    this.mountPoint = mountPoint;
+  }
+
+  initMaterialCard() {
```
Comment (ihor4568) [inline src/components/Albums/AlbumListItemComponent/AlbumListItemComponent.js]: it should be called initMaterial

File: src/index.scss [hunk 7]
```diff
@@ -9,3 +12,4 @@
 @import "./components/Main/Main";
 @import "./components/Player/Player";
 @import "./components/Page/Page";
+@import "components/Albums/albumsComponent";
```
Comment (ihor4568) [inline src/index.scss]: why file name is in lower case?

**Comment by ihor4568** [comment]
Please do smth to naming of files and styles

* Each folder should be named after component name staring with capital letter
* There should be 3 files - js html and scss inside
* WE ARE USING BEM - thus your classes in styles should be named properly - name of block is lowercase name of the component

File: src/index.scss (hunk 7 above)
Comment (OlegShynkarenko) [reply src/index.scss]: Corrected

File: src/components/Albums/AlbumListItemComponent/AlbumListItemComponent.js (hunk 6 above)
Comment (OlegShynkarenko) [reply src/components/Albums/AlbumListItemComponent/AlbumListItemComponent.js]: Corrected

File: src/components/Albums/albumsComponent.scss (hunk 3 above)
Comment (OlegShynkarenko) [reply src/components/Albums/albumsComponent.scss]: Corrected

File: src/components/Albums/AlbumListItemComponent/AlbumListItem.html (hunk 4 above)
Comment (OlegShynkarenko) [reply src/components/Albums/AlbumListItemComponent/AlbumListItem.html]: done

File: .firebase/hosting.ZGlzdA.cache [hunk 8]
```diff
@@ -0,0 +1,3 @@
+index.html,1540845529969,dcb1dc9d2cb2a3e57d52aa0c1f3524a5c0c75fcae9b1c2cb96a33dad07035cf2
```
Comment (ihor4568) [inline .firebase/hosting.ZGlzdA.cache]: ?? what is it?

File: src/components/Albums/AlbumCard/AlbumCard.js [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+        sourceImage: ""
+      }
+    }
+  ) {
+    this.mountPoint = mountPoint;
+    this.props = props;
+  }
+
+  initMaterial() {
+    const cardRipple = this.mountPoint.querySelectorAll(
```
Comment (ihor4568) [inline src/components/Albums/AlbumCard/AlbumCard.js]: this should be inside querySelectors

File: src/components/Albums/AlbumCard/AlbumCard.js [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+      ".album-card__card-ripple-effect"
+    );
+    Array.from(cardRipple).forEach(el => {
+      new MDCRipple(el);
+    });
+  }
+
+  setAlbumsDescriptions() {
+    this.props = {
```
Comment (ihor4568) [inline src/components/Albums/AlbumCard/AlbumCard.js]: please do not reassign props. Why do you need to do this?

File: src/components/Albums/AlbumCard/AlbumCard.js [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+      albumName: this.props.album.albumName,
+      artistName: this.props.album.artistName,
+      sourceImage: this.props.album.sourceImage
+    };
+  }
+
+  mount() {
+    this.mountPoint.innerHTML += this.render();
```
Comment (ihor4568) [inline src/components/Albums/AlbumCard/AlbumCard.js]: why are you doing `+= render()` ??

File: src/components/Albums/AlbumCard/AlbumCard.js [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+    this.initMaterial();
+  }
+
+  render() {
+    this.setAlbumsDescriptions();
```
Comment (ihor4568) [inline src/components/Albums/AlbumCard/AlbumCard.js]: you should not do this in render

File: src/components/Albums/AlbumList/AlbumList.js [hunk 13]
```diff
@@ -0,0 +1,96 @@
+import { AlbumCard } from "../AlbumCard/AlbumCard";
+import albumListTemplate from "./AlbumList.html";
+
+export class AlbumList {
+  constructor(mountPoint, props = []) {
```
Comment (ihor4568) [inline src/components/Albums/AlbumList/AlbumList.js]: props should be an object

File: src/components/Albums/AlbumList/AlbumList.js [hunk 14]
```diff
@@ -0,0 +1,96 @@
...
+  querySelectors() {
+    this.albumsList = this.mountPoint.querySelector(".album-list__container");
+  }
+
+  mountChildren() {
+    const albums = [
+      {
+        albumName: "Loud",
+        artistName: "by Rihanna",
+        sourceImage: 'src="./assets/img/AlbumsCovers/rihanna_loud.jpg"'
```
Comment (ihor4568) [inline src/components/Albums/AlbumList/AlbumList.js]: really? 'src="./assets/img/AlbumsCovers/rihanna_loud.jpg"' ?? You should not have this `src=` prefix in there

File: src/components/Albums/AlbumList/AlbumList.js [hunk 15]
```diff
@@ -0,0 +1,96 @@
...
+    this.mountPoint = mountPoint;
+    this.props = props;
+  }
+
+  querySelectors() {
+    this.albumsList = this.mountPoint.querySelector(".album-list__container");
+  }
+
+  mountChildren() {
+    const albums = [
```
Comment (ihor4568) [inline src/components/Albums/AlbumList/AlbumList.js]: this albums should have been received from props

File: src/components/Albums/AlbumCard/AlbumCard.scss [hunk 16]
```diff
@@ -0,0 +1,21 @@
+.album-card {
+  margin: 0 0.9rem 2.5rem 0.9rem;
```
Comment (ihor4568) [inline src/components/Albums/AlbumCard/AlbumCard.scss]: the component should not know about its positioning. Please move this margin top parent component

File: src/components/Main/Main.js [hunk 17]
```diff
@@ -36,6 +37,9 @@ export class MainComponent {
     if (e.target.closest(".main__about-link")) {
       this.about.mount();
     }
+    if (e.target.closest(".main__albums-link")) {
```
Comment (ihor4568) [inline src/components/Main/Main.js]: else if

**Comment by ihor4568** [comment]
please fix this 
image

File: .firebase/hosting.ZGlzdA.cache (hunk 8 above)
Comment (OlegShynkarenko) [reply .firebase/hosting.ZGlzdA.cache]: It was firebase hosting cashe folder. I put it in .gitignore file.

File: src/components/Albums/AlbumCard/AlbumCard.js (hunk 9 above)
Comment (OlegShynkarenko) [reply src/components/Albums/AlbumCard/AlbumCard.js]: Done

File: src/components/Albums/AlbumCard/AlbumCard.js (hunk 11 above)
Comment (OlegShynkarenko) [reply src/components/Albums/AlbumCard/AlbumCard.js]: It was needed to display all albums from Albums array. Currently I've merged components in the one structure and use ejs.

File: src/components/Main/Main.js (hunk 17 above)
Comment (OlegShynkarenko) [reply src/components/Main/Main.js]: Corrected

File: src/components/Albums/AlbumList/AlbumList.js (hunk 14 above)
Comment (OlegShynkarenko) [reply src/components/Albums/AlbumList/AlbumList.js]: Corrected

File: src/components/Albums/AlbumList/AlbumList.js (hunk 13 above)
Comment (OlegShynkarenko) [reply src/components/Albums/AlbumList/AlbumList.js]: Corrected

File: src/components/Albums/AlbumCard/AlbumCard.scss (hunk 16 above)
Comment (OlegShynkarenko) [reply src/components/Albums/AlbumCard/AlbumCard.scss]: Corrected

File: src/components/Albums/AlbumCard/AlbumCard.js (hunk 12 above)
Comment (OlegShynkarenko) [reply src/components/Albums/AlbumCard/AlbumCard.js]: Thanks for your note. Currently this method was removed and components structure has been changed.

File: src/components/Albums/Albums.js [hunk 18]
```diff
@@ -0,0 +1,96 @@
...
+    imageSource: "https://image.ibb.co/jTzR2f/ariannagrande-myeverything.jpg"
+  },
+  {
+    albumName: "Divide",
+    artistName: "by Ed Sheeran",
+    imageSource: "https://image.ibb.co/iKPha0/edsheeran-divide.jpg"
+  }
+];
+
+export class Albums {
```
Comment (ihor4568) [inline src/components/Albums/Albums.js]: this should be called AlbumsComponent

File: src/components/Page/page.scss [hunk 19]
```diff
@@ -0,0 +1,3 @@
+.page {
```
Comment (ihor4568) [inline src/components/Page/page.scss]: Page.scss should start with uppercase letter

File: src/index.scss [hunk 20]
```diff
@@ -13,5 +16,6 @@
 @import "./components/Main/Main";
 @import "./components/Player/Player";
 @import "./components/Page/Page";
+@import "components/Albums/Albums";
```
Comment (ihor4568) [inline src/index.scss]: "./components/.." - relative path

File: src/components/Albums/Albums.js (hunk 18 above)
Comment (OlegShynkarenko) [reply src/components/Albums/Albums.js]: Fixed

File: src/components/Page/page.scss (hunk 19 above)
Comment (OlegShynkarenko) [reply src/components/Page/page.scss]: Fixed

File: src/index.scss (hunk 20 above)
Comment (OlegShynkarenko) [reply src/index.scss]: Fixed

**Review by gamesminer — APPROVED** [review]

**Pull request merged**